src/gt4py/next/embedded/common.py

```diff
@@ -148,7 +148,9 @@ def restrict_to_intersection(
     )
 
 
-def iterate_domain(domain: common.Domain):
+def iterate_domain(
+    domain: common.Domain,
+) -> Generator[tuple[tuple[common.Dimension, int], ...], None, None]:
```

Comment:
Just a question: would it be enough with `Iterator` instead of `Generator`?  In any case, use the actual ABCs from `collections.abc` instead of the deprecated aliases in `typing`

Reply:
changed to Iterator